crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -369,9 +367,8 @@ def proposal_update
       return render :text => ret[1], :status => ret[0] if ret[0] != 200
       return render :json => ret[1]
     else # This is UI.
-      params[:id] = "bc-#{params[:barclamp]}-#{params[:id] || params[:name]}"
       if params[:submit] == t('barclamp.proposal_show.save_proposal')
-        @proposal = ProposalObject.find_proposal_by_id(params[:id])
+        @proposal = Proposal.where(barclamp: params[:barclamp], name: params[:id] || params[:name]).first
```

Comment:
I'm a bit unsure here: it seems `params[:name]` would be the full name (`bc-barclamp-foo`), in which case I don't think it's correct to use it that way.